### ISSUE TITLE
b.map is not a function error when LDAP server is offline

### ISSUE THREAD

**Issue by exdolyeg** [keycloak/keycloak#24293]
### Before reporting an issue

- [X] I have searched existing issues
- [X] I have reproduced the issue with the [latest nightly release](https://github.com/keycloak/keycloak/releases/tag/nightly)

### Area

core

### Describe the bug

The following error is generated if an LDAP provider User federation is offline.  This error shows up when trying to list all users.

![Keycloak LDAP offline Error](https://github.com/keycloak/keycloak/assets/59295730/c9c99815-ad9b-4dd8-ae29-0892b1683e7d)

### Version

22.0.5

### Expected behavior

Local users and users from other LDAP providers should be listed.  Users from the offline Federation should be listed but can be tagged as offline.

### Actual behavior

The error in the screenshot above is displayed.  Clicking the "Press here to refersh and continue" takes you  back tot he master realm roles tab

### How to Reproduce?

1. Configure a working LDAP provider in User Federation. I used both Active Directory and OpenLDAP.  These work perfectly once keycloak can connect to the these LDAP services.
2. Take one or both LDAP service offline
3. Navigate to Users tab and Search for all user by supplying "*"

### Anything else?

My work-around is to disable the LDAP provider that is offline.
Once that is done all users both local and from other online providers are returned when searching.

**Comment by jonkoops** [comment]
Do you have any server logs and browser logs that you could share? If you open the browser console you should be able to get a more accurate stack trace.

**Comment by jonkoops** [comment]
Possible duplicate of #24138

**Comment by exdolyeg** [comment]
![Browser_console 2023-10-25 132830](https://github.com/keycloak/keycloak/assets/59295730/2ea4fe9a-e762-4731-adbf-e37c07dc4734)

**Comment by exdolyeg** [comment]
> Possible duplicate of #24138

The behaviour is the same whether I use name, email address or "*" to search.

**Comment by exdolyeg** [comment]
I think it is more related to this issue https://github.com/keycloak/keycloak/issues/23396.

This is the behavior for v22.0.1 to v22.0.4

**Comment by rmartinc** [comment]
@jonkoops I think this issue is what it was commented in the previous issue https://github.com/keycloak/keycloak/issues/23396#issuecomment-1755205115. If the rest call returns an error: `BAD_REQUEST 400 -> {"error":"unknown_error"}`. The admin console fails with `b.map is not a function`. You can test just defining an ldap server that doesn't exists (or stopping it after configuring it). I suppose the page should display the error and display no users.

@exdolyeg Returning the error is the current server side behavior for the search endpoint. You can disable any failing LDAP provider to not search on that server.

**Comment by exdolyeg** [comment]
@rmartinc - I know that disabling the failing LDAP provider resolves the error, however is this is expected solution, or is this being treated as a bug?

**Comment by jonkoops** [comment]
Yeah, we need to handle errors here in a better manner, I am not sure why this doesn't show the proper error state in the UI. We'll have to get to the bottom of that.

**Comment by jonkoops** [comment]
@ssilvert I gave this a medium impact, feel free to change this if you disagree about the severity.
